corehq/apps/users/static/users/js/mobile_workers.ng.js

```diff
@@ -0,0 +1,201 @@
+
+    mobileWorkerControllers.MobileWorkerCreationController = function (
+            $scope, workerCreationFactory, djangoRMI, customFields,
+            customFieldNames
+    ) {
+        $scope._ = _;  // make underscore available
+        $scope.mobileWorker = {};
+        $scope.usernameAvailabilityStatus = null;
+        $scope.usernameStatusMessage = null;
+        $scope.workers = [];
+        $scope.customFormFields = customFields;
+        $scope.customFormFieldNames = customFieldNames;
+
+        $scope.initializeMobileWorker = function (mobileWorker) {
+            visualFormCtrl.usernameClear();
+            $scope.usernameAvailabilityStatus = null;
+            $scope.usernameStatusMessage = null;
+
+            if (!_.isEmpty(mobileWorker)) {
+                mobileWorker.creationStatus = STATUS.RETRIED;
+                $scope.mobileWorker = new MobileWorker({
+                    customFields: mobileWorker.customFields,
+                    username: mobileWorker.username
+                });
+            } else {
+                $(".select2multiplechoicewidget").select2('data', null);
+                $scope.mobileWorker = new MobileWorker({customFields: customFields});
+            }
+        };
+
+        $scope.submitNewMobileWorker = function () {
+            $("#newMobileWorkerModal").modal('hide');
+            $scope.workers.push($scope.mobileWorker);
+            workerCreationFactory.stageNewMobileWorker($scope.mobileWorker);
+        };
+    };
+
+    var mobileWorkerFactories = {};
+    mobileWorkerFactories.workerCreationFactory = function ($q, djangoRMI) {
+        var self = {};
+
+        self.stageNewMobileWorker = function (newWorker) {
+            newWorker.creationStatus = STATUS.PENDING;
+            var deferred = $q.defer();
+            djangoRMI.create_mobile_worker({
+                mobileWorker: newWorker
+            })
+            .success(function (data) {
+                if (data.success) {
+                    newWorker.creationStatus = STATUS.SUCCESS;
+                    newWorker.editUrl = data.editUrl;
+                    deferred.resolve(data);
+                } else {
+                    newWorker.creationStatus = STATUS.WARNING;
+                    deferred.reject(data);
+                }
+            })
+            .error(function () {
+                newWorker.creationStatus = STATUS.WARNING;
+                deferred.reject(
+                    "Sorry, there was an issue communicating with the server."
+                );
+            });
+
+            return deferred.promise;
+        };
+        return self;
+    };
+
+    var mobileWorkerDirectives = {};
+    mobileWorkerDirectives.validateUsername = function ($http, $q, djangoRMI) {
+        return {
+            restrict: 'AE',
+            require: 'ngModel',
+            link: function ($scope, $elem, $attr, ctrl) {
+                ctrl.$validators.validateUsername = function (username) {
```

Comment:
Is this where this validator is hooked in?  It's added to some list of validators taht get called on `ngvalidate`?  (Trying to check my understanding)